source/operation.ts

```diff
@@ -45,12 +45,32 @@ export interface DeleteOperation {
   entity_key: string[];
 }
 
+export interface QueryServerInformationOperation {
+  action: "query_server_information";
+  values?: string[];
+}
+
+export interface QuerySchemasOperation {
+  action: "query_schemas";
+}
+
+export interface GetUploadMetadataOperation {
+  action: "get_upload_metadata";
+  file_name: string;
+  file_size: number;
+  component_id: string;
+}
+
 export type Operation =
   | CreateOperation
   | QueryOperation
   | SearchOperation
   | UpdateOperation
-  | DeleteOperation;
+  | DeleteOperation
+  | QueryServerInformationOperation
+  | QuerySchemasOperation
+  | GetUploadMetadataOperation
+  | { action: string; [key: string]: any };
```

Comment:
I really didn't want to add this line since this picks up all other cases, e.g. `{ action: "create", wrong: "type" }`, but it was the only way i could think of to have a generic operation without having to list (and maintain) all valid action types. I tried to come up with a way to specify a type that is "every string except already specified action values", but looks like negated types like that isn't supported in typescript yet.

unless you come up with a better solution. @lucaas @gismya @WaylonFtrack

Reply:
tried with something like this:
```ts
type Key = TypedOperation['action'];
type NegateKey = Exclude<string, Key>;
type OtherOperation = { action: NegateKey; [key: string]: any };
```

no luck. the `NegateKey` will be type `string`.